src/pocketmine/utils/Utils.php

```diff
@@ -547,4 +547,13 @@ public static function errorExceptionHandler(int $severity, string $message, str
 
 		return true; //stfu operator
 	}
+
+	public static function validateObjectArray(array $array, string $class) : bool{
+		foreach($array as $key => $item){
+			if(!($item instanceof $class)){
+				throw new \TypeError("Element \"$key\" is not an instance of $class");
+			}
+		}
+		return true;
```

Comment:
note to self: document the apparently-useless return value here (it's for `assert()`).

This change should be separated from the other changes since it's not directly related.